src/appshell/view/preferences/generalpreferencesmodel.cpp

```diff
@@ -54,30 +54,19 @@ void GeneralPreferencesModel::load()
 
 void GeneralPreferencesModel::checkUpdateForCurrentLanguage()
 {
-    QString currentLanguageCode = this->currentLanguageCode();
-    LanguageStatus::Status languageStatus = languagesService()->languageStatus(currentLanguageCode);
-    if (languageStatus != LanguageStatus::Status::NeedUpdate) {
-        Language currentLanguage = languagesService()->languages().val.value(currentLanguageCode);
-        QString msg = mu::qtrc("appshell/preferences", "Your version of %1 is up to date").arg(currentLanguage.name);
-        interactive()->info(msg.toStdString(), "");
-        return;
-    }
+    QString currentLanguageCode = effectiveCurrentLanguageCode();
```

Comment:
effective? What does it mean? :)

Reply:
The specified language code (for example, "system") needs to be transformed into a code for which we provide a translation (for example, "en_US" or "nl"). The latter is what I call the "effective" language code.

Reply:
This "efficient" term is very confusing. Why can't LanguagesService::currentLanguageCode() return the code in the proper format?

Reply:
`effective`, not `efficient`. I believe the word `effective` has a secondary meaning like "actual", "being used", "in fact"; that is the meaning I'm referring to in this case. 

We need both the specified code and the "actual" code: when determining which item should be selected in the list in Preferences, we need the specified code (otherwise, we could not show the user that the `"system"` language is selected), and when doing any operation with LanguagesService, we need the "actual" code, because `"system"` is not a language itself.

Would it be clearer to call the method `actualLanguageCode`, or `realLanguageCode`, or `usableLanguageCode`? Personally I think none of those is really better...

Reply:
> We need both the specified code and the "actual" code: when determining which item should be selected in the list in Preferences, we need the specified code (otherwise, we could not show the user that the "system" language is selected), and when doing any operation with LanguagesService, we need the "actual" code, because "system" is not a language itself.

I think, only service should know about this logic. 

The logic is identical to audio and midi device selection. There, the model is very simple - get from the service a list of all devices(contains a system default) and the current device(might be a system default) and displays the current state.
The selected device is transferred to the configurator and written to the settings. The service subscribes to changing the current device from the configurator and connects to the selected device.

That is, model and configuration does not know about the logic with the system default. All logic is placed in the service.